### PR TITLE
Bump version to 6.0.1 on the 6.x branch

### PR DESCRIPTION
**Pull request by DaveCTurner**
Mainly opening this PR to get a full CI run before merging, as it's mostly a port of #27386 to the 6.x branch. The changes to `VersionUtils#resolveReleasedVersions()` and associated tests deserve attention, and ultimately backporting to 6.0 too.

I'd wait for CI to pass before looking at this too hard.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
CI passed, so a review of my changes to `VersionUtils` would be appreciated.

File: core/src/test/java/org/elasticsearch/VersionTests.java [hunk 1]
```diff
@@ -343,8 +343,6 @@ public void testIsCompatible() {
         } else {
             assertFalse(isCompatible(Version.CURRENT, Version.fromString("7.0.0")));
         }
-        assertFalse("only compatible with the latest minor",
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/VersionTests.java]: why do we remove this assertion, I think it's valid?

**Review by s1monw — CHANGES_REQUESTED** [review]
left one question LGTM otherwise

File: core/src/test/java/org/elasticsearch/VersionTests.java (hunk 1 above)
Comment (DaveCTurner) [reply core/src/test/java/org/elasticsearch/VersionTests.java]: Sorry, yes, you're right it's valid. There's a bug somewhere. Hunting it now...

**Comment by DaveCTurner** [comment]
@simonw ok, assertion reinstated and fixed with a change to `Version.minimumCompatibilityVersion`, and the conflicting `assertTrue(isCompatible(Version.fromString("6.0.0"), Version.fromString("7.0.0")));` removed instead. Could you take another look?

**Review by s1monw — APPROVED** [review]
LGTM

**Review by jasontedor — COMMENTED** [review]
LGTM.

**Pull request merged**